AzUtilities/AzBlob.py

```diff
@@ -0,0 +1,265 @@
+
+    def __init__(self, account_name, account_key):
+        """
+
+        :param account_name: name of the blob account
+        :param account_key: key of the blob account
+        """
+        self.block_blob_service = BlockBlobService(account_name=account_name, account_key=account_key)
+        self.append_blob_service = AppendBlobService(account_name=account_name, account_key=account_key)
+        self.name = None
+        self.mode = None
+        self.f = None
+        self.container = None
+        self.blob = None
+        # FIXME I do not fully understand + here. Might be redundant for us
+        self._read_modes = set(['r', 'rb', 'r+b'])
+        self._write_modes = set(['w', 'wb', 'w+b'])
+        self._append_modes = set(['a', 'ab', 'a+b'])
+        self._appendblob_modes = set(['blobat', 'blobab', 'blobap', 'blobas'])
+
+    def _get_temp_path(self):
+        tmp_file = tmp.TemporaryFile()
+        self.name = tmp_file.name  # tmp.tempdir + '/tmp' + ''.join(np.random.choice(['a', 'v', 'b', 'c', 'm', 'n'], 5))
+        tmp_file.close()
+
+    def _open_read_or_append(self):
+        self._get_temp_path()
+        self.block_blob_service.get_blob_to_path(container_name=self.container,
+                                                 blob_name=self.blob,
+                                                 file_path=self.name)
+        self.f = open(self.name, mode=self.mode)
+
+    def _open_write_mode(self):
+        files_set = self.list_files(self.container, do_print=False, as_set=True)
+        if self.blob in files_set:
+            warnings.warn('The blob already exist in the container. Will be overwritten if you close. '
+                          'Run delete_temp() to delete the temp file without uploading')
+            # answer = input('The blob already exist in the container. Would you like to overwrite it? (y/N)', end='')
+            # answer = sys.stdin.read()
+            # print('')
+            # if answer != 'y':
+            #     return
+        self.name = tmp.gettempdir() + '/' + self.blob
+        self.f = open(self.name, self.mode)
+
+    def _open_append_blob(self, container, blob):
+        self._append_container = container
+        self._append_blob = blob
+
+    def open(self, container, blob, mode='r'):
+        """
+        Open a file like object and return it if it was opened in read mode. The ideal was not to return that and
+        only work with the class. But it did not work with pd.read_csv and json.load. So the output of them should
+        be feed to these functions instead. **remeber to close the class not the output of this function** if you
+        do the temporary file would remain on the system.
+
+        :param container:
+        :param blob:
+        :param mode: str
+            Will feed to open()
+        :return:
+        """
+        if self.f is not None:
+            self.close()
```

Comment:
What does this mean? If you use `open` twice, it will close the old file before opening it again?